crypto/crypto/src/ed25519.rs

```diff
@@ -214,6 +218,21 @@ impl SigningKey for Ed25519PrivateKey {
     type VerifyingKeyMaterial = Ed25519PublicKey;
     type SignatureMaterial = Ed25519Signature;
 
+    fn sign<T: CryptoHash + Serialize>(
+        &self,
+        message: &T,
+    ) -> Result<Ed25519Signature, CryptoMaterialError> {
+        let mut bytes = Vec::<u8>::new();
+        bytes.extend_from_slice(&<T::Hasher as CryptoHasher>::salt().as_bytes());
```

Comment:
The salt is an un-hashed type name? Shouldn't it be hashed here?

Reply:
I guess, we want salts to be pre-hashed already, so we don't hash them over and over again...

BTW, if we had a suffix based separator (like the one we had before - something like $$$###LIBRA###$$$ that will clearly not be part of any type name) we might not need hashing of salts, they will be guaranteed to not be prefixes of each other. But we will loose the property of salts being of the same length in case we need it.

Reply:
See my previous comment: I believe we should inject a fixed-length (i.e. hashed) seed here instead of the salt.

Reply:
The suffix of the salt is now a fixed prefix. It was never important for security as we were hashing the salt anyway (partly for performance reasons).

Reply:
on an unrelated note, this seems like we're creating to Vec<u8>

why not make it

```
let mut bytes = <T::Hasher as CryptoHasher>::salt().as_bytes();
```
or maybe appending an additional to_vec after that?

Reply:
As argued (with a ton of sense) by @matbd, this now injects a 32-byte sha3-seed. And I added @davidiw 's `to_vec()`.